core-plugins/src/test/java/io/cdap/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -508,45 +510,56 @@ public void testReadJson() throws Exception {
 
   @Test
   public void testReadCSV() throws Exception {
-    testReadDelimitedText(FileFormat.CSV.name(), ",");
+    testReadDelimitedText(FileFormat.CSV.name(), ",", false);
+    testReadDelimitedText(FileFormat.CSV.name(), ",", true);
   }
 
   @Test
   public void testReadTSV() throws Exception {
-    testReadDelimitedText(FileFormat.TSV.name(), "\t");
+    testReadDelimitedText(FileFormat.TSV.name(), "\t", false);
+    testReadDelimitedText(FileFormat.TSV.name(), "\t", true);
   }
 
   @Test
   public void testReadDelimited() throws Exception {
-    testReadDelimitedText(FileFormat.DELIMITED.name(), "\u0001");
+    testReadDelimitedText(FileFormat.DELIMITED.name(), "\u0001", false);
+    testReadDelimitedText(FileFormat.DELIMITED.name(), "\u0001", true);
   }
 
-  private void testReadDelimitedText(String format, String delimiter) throws Exception {
+  private void testReadDelimitedText(String format, String delimiter, boolean enableQuotedValues) throws Exception {
     File fileText = new File(temporaryFolder.newFolder(), "test.txt");
     String outputDatasetName = UUID.randomUUID().toString();
 
     Schema schema = Schema.recordOf("user",
                                     Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
-                                    Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                                    Schema.Field.of("val1", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                                    Schema.Field.of("val2", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                                    Schema.Field.of("val3", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
                                     Schema.Field.of("file", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
 
     String appName = UUID.randomUUID().toString();
     ApplicationManager appManager = createSourceAndDeployApp(appName, fileText, format, outputDatasetName, schema,
-                                                             delimiter);
+                                                             enableQuotedValues, delimiter);
 
+    String join = Joiner.on(delimiter).join(new String[] {"\"a", "b", "c\""});
```

Comment:
can we also test multiple quoted values on the same line, something like:

"a, b", "c, d"

to make sure it grabs "a, b" as one value and "c, d" as another. We shouldn't have to run a pipeline to test this type of logic, let's do some refactoring so that these types of unit tests can be done efficiently.